File: leetcode/problems_list/1-TwoSum/solution.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 *
 * Time Complexity - O(n) - Linear
 * Space Complexity - O(n) - Linear
 */
var twoSum = function (nums, target) {
  let mappedValues = {},
    pairIdx = [];

  nums.forEach((curr, idx) => {
    if (target - curr in mappedValues)
      pairIdx.push(idx, mappedValues[target - curr]);
    mappedValues[curr] = idx;
  });

  return pairIdx;
};

File: leetcode/problems_list/1-TwoSum/solution.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        mapped_values, pair_idx = dict(), []
        for idx, curr in enumerate(nums):
            if (target-curr in mapped_values):
                pair_idx = [idx, mapped_values[target-curr]]
            mapped_values[curr] = idx
        return pair_idx
